### ISSUE TITLE
Sorted neighbor list

### ISSUE THREAD

**Issue by pastewka** [libAtoms/matscipy#50]
The neighbor list is currently sorted by first index only, it may be beneficial for some applications to have a fully sorted neighbor list. This would make neighbor list construction slower, but only slightly so. Any opinions on this?

**Comment by jameskermode** [comment]
Maybe optional and defaulting to not sorting?

**Comment by griessej** [comment]
I agree with James. If this is needed, it should be passed explicitly by keyword. Applications that do not need this will not suffer any performance loss.

**Comment by pastewka** [comment]
That would be an option. I'll look into this.
